### PR TITLE
Bump ui_test to 0.22

### PR DESCRIPTION
**Pull request by oli-obk**
*Please write a short comment explaining your change (or "none" for internal only changes)*

changelog: none

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]

**Comment by Manishearth** [comment]
r=me but tests fail

File: tests/ui/absurd-extreme-comparisons.stderr [hunk 1]
```diff
@@ -1,5 +1,5 @@
 error: this comparison involving the minimum or maximum element for this type contains a case that is always true or always false
-  --> $DIR/absurd-extreme-comparisons.rs:14:5
+  --> tests/ui/absurd-extreme-comparisons.rs:14:5
```
Comment (flip1995) [inline tests/ui/absurd-extreme-comparisons.stderr:2]: Those changes look like they will cause trouble in the Rust repo. :thinking: I haven't tested that yet, though.

Comment (oli-obk) [reply tests/ui/absurd-extreme-comparisons.stderr]: I changed it so it doesn't show the tests/ui part, but it still shows the subdirectories within that

**Comment by oli-obk** [comment]
and now with the 0.22.1 release, the proc macro changes are not needed anymore

**Pull request merged**